### PR TITLE
Gasmixture share() gets much better

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
image

## Why It's Good For The Game
image

### CODE REVIEW

**Comment by willox** [comment]
If you're going to keep `/datum/gas_mixture/proc/garbage_collect` around, maybe it should be implemented using the macro to avoid the code being duplicated?

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -8,6 +8,11 @@ This prevents race conditions that arise based on the order of tile processing.
 GLOBAL_LIST_INIT(meta_gas_info, meta_gas_list()) //see ATMOSPHERICS/gas_types.dm
 GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
+#define ATMOS_GARBAGE_COLLECT(gaslist, tocheck)\
+	for(var/datum/gas/id as anything in (tocheck || gaslist)){\
+		if(QUANTIZE(gaslist[id][MOLES]) <= 0) gaslist -= id;\
+	}
+
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm:15]: This isn't a valid change. Garbage collect is overriden by code that needs to be able to react to it.
We cannot move it to a macro

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by LemonInTheDark** [comment]
The self cpu reduction does not make a lot of sense. I'd expect total to reduce, and self to go up slightly. Something about your profile is weird, it may be a rut or something.

These garbage collects are also not part of the hot path here

**Comment by LemonInTheDark** [comment]
thank you for trying though, I'm very happy someone other then me is looking at this stuff

**Comment by LemonInTheDark** [comment]
For the record if you wanted to optimize garbage collect, split the collecting one list and collecting the whole gasmix behavior into separate procs. I'm not sure how big a difference it would functionally make tho

**Comment by Ghilker** [comment]
Will try again to make it better

**Pull request closed without merging**